jupyterhub/auth.py

```diff
@@ -635,6 +635,24 @@ def delete_user(self, user):
         """
         self.allowed_users.discard(user.name)
 
+    manage_groups = Bool(
+        config=True, help="Let authenticator manage user groups"
+    )
+
+    @default('manage_groups')
+    def _manage_groups_default(self):
+        return False
+
+    def load_user_groups(self, auth_state):
+        """Hook called allowing authenticator to read user groups
+
+        Updates user group memberships
+
+        Args:
+            auth_state (dict): Propriatary dict returned by authenticator
```

Comment:
We should specify: this should _return_ a list of group names. Maybe even call it `get_user_groups(user, auth_state)`.